python/lsst/pipe/tasks/makeDirectWarp.py

```diff
@@ -0,0 +1,799 @@
+    ConfigField,
+    ConfigurableField,
+    Field,
+    FieldValidationError,
+    ListField,
+    RangeField,
+)
+from lsst.pipe.base import (
+    Instrument,
+    NoWorkFound,
+    PipelineTask,
+    PipelineTaskConfig,
+    PipelineTaskConnections,
+    Struct,
+)
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+from lsst.pipe.tasks.makeWarp import reorderRefs
+from lsst.skymap import BaseSkyMap
+from lsst.utils import inheritDoc
+
+from .coaddInputRecorder import CoaddInputRecorderTask
+
+if TYPE_CHECKING:
+    from lsst.afw.image import MaskedImage
+    from lsst.afw.table import Exposure, ExposureCatalog
+
+
+log = logging.getLogger(__name__)
+
+__all__ = (
+    "MakeDirectWarpConfig",
+    "MakeDirectWarpTask",
+)
+
+
+class MakeDirectWarpConnections(
+    PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap", "instrument", "visit"),
+    defaultTemplates={
+        "coaddName": "deep",
+        "calexpType": "",
+    },
+):
+    """Connections for MakeWarpTask"""
+
+    calexp_list = Input(
+        doc="Input exposures to be interpolated, resampled and optionally PSF-matched onto a SkyMap "
+        "projection/patch",
+        name="{calexpType}calexp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    background_list = Input(
+        doc="Input background to be added back into the calexp if bgSubtracted=False",
+        name="calexpBackground",
+        storageClass="Background",
+        dimensions=("instrument", "visit", "detector"),
+        multiple=True,
+    )
+    skyCorr_list = Input(
+        doc="Input Sky Correction to be subtracted from the calexp if doApplySkyCorr=True",
+        name="skyCorr",
+        storageClass="Background",
+        dimensions=("instrument", "visit", "detector"),
+        multiple=True,
+    )
+    visit_summary = Input(
+        doc="Input visit-summary catalog with updated calibration objects",
+        name="finalVisitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit"),
+    )
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs for warped exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    # Declare all possible outputs (except noise, which is configurable)
+    direct_warp = Output(
+        doc="Output direct warped exposure produced by resampling calexps onto the skyMap patch geometry",
+        name="{coaddName}Coadd_directWarp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "instrument", "visit"),
+    )
+    direct_maskedfraction_warp = Output(
+        doc="Output masked fraction warped exposure",
+        name="{coaddName}Coadd_directWarp_maskedFraction",
+        storageClass="MaskedImageF",
```

Comment:
This can totally be persisted as an `ImageF`. In memory, it should remain a `MaskedImageF` given [DM-42948](https://jira.lsstcorp.org/browse/DM-42948) but I don't see a reason to persist it as a `MaskedImageF`.